lib/logstash/outputs/elasticsearch/http_client.rb

```diff
@@ -30,19 +30,34 @@ def bulk(actions)
       bulk_body = actions.collect do |action, args, source|
         if action == 'update'
```

Comment:
Have you considered breaking up this if block into a separate method? The method length here is rather large.